src/enochecker/utils.py

```diff
@@ -319,4 +348,19 @@ def write(self, buffer: Union[str, bytes]) -> None:
 
         :param buffer: The buffer to write
         """
-        super().write(ensure_bytes(buffer))
+        super().send(ensure_bytes(buffer))
+
+    def readn(self, size: int, timeout: Optional[float] = None) -> bytes:
+        """
+        Read n bytes from the socket.
+
+        Can block if the connection is blocked.
+        May raise socket.error if the connection is closed.
+
+        :param size: Number of bytes to read
+        :param timeout: Timeout for read operation
+        """
+        if timeout is None:
+            timeout = self.current_default_timeout
+
+        return super().recvn(size, timeout=timeout)
```

Comment:
true! I'll remove that

Reply:
It is via the 'timeout' keyword arg! 

I actually noticed that every 'recv' function can be called via a corresponding 'read', same goes for `send` and `write`. Here is the code in question:

```python3
for _name in list(locals()):
    if 'recv' in _name:
        _name2 = _name.replace('recv', 'read')
    elif 'send' in _name:
        _name2 = _name.replace('send', 'write')
    else:
        continue
    locals()[_name2] = make_wrapper(locals()[_name], _name2)
```

The rest of our wrappers are still needed though because of our use of snake_case.

Reply:
Ah, I was talking about `self.current_default_timeout` if we want to support this

Reply:
Then should I revert that last commit?

Reply:
how would we enforce the timeout? on the worker-level?

Reply:
Originally we calculated remaining time for each request, based on the total available time. I guess this is no longer strictly needed as the engine checks timeouts, too